CreeDictionary/utils/paradigm_filler.py

```diff
@@ -100,11 +136,13 @@ def fill_paradigm(
                     if isinstance(cell, StaticCell) or cell == "":
                         # We do nothing to static and empty cells.
                         continue
-
-                    # It's a inflection form pattern
-                    assert '"' not in cell
-                    lookup_strings.append(cell.replace("{{ lemma }}", lemma))
-                    string_locations.append((row_with_replacements, col_ind))
+                    elif isinstance(cell, InflectionCell):
+                        lookup_strings.append(
+                            ConcatAnalysis(cell.analysis.substitute(lemma=lemma))
```

Comment:
A minor refactor to consider: to avoid violating [the Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter#In_object-oriented_programming), you coud add a method to `InflectedCell` that would do `return ConcatAnalysis(self.analysis.substitute(lemma=lemma))`, and then this line would be:

```suggestion
                            cell.create_concat_analysis_for(lemma))
```

I'm not super sure about the name though 😅 

---

This would make things slightly easier to change if we ever decide to change how the string template is implemented — we wouldn't need to know that analysis is a `string.Template` in this code; only in the `InflectedCell` code.

Reply:
yes I had a `fill` method and I deleted it lol idk why. Sure thing!